doc/style-guide-testing.md

```diff
@@ -0,0 +1,467 @@
+
+
+### Test Behavior
+
+We want to write tests to not only confirm there are no defects, but also ones that could mimic how the software is intended to be used.  By testing behavior, we are shifting our viewpoint to be from the user's point of view instead of being overly concerned with the internals of the software.  In essence, I want to make sure your code performs a specific action to produce the desired result.  I do not want to guarantee that the code does every specific step in exactly the right order.
+
+When you test behavior, you should be able to completely rewrite the internals of how the job gets done and the tests should all pass.  Depend on inputs and outputs.  Pretend the code is opaque.
+
+Angular directives that emit events, for instance, should have tests that listen on the parent scope.  Listening on the directive's scope is not correct because those listeners would be triggered by a `.$broadcast()` as well and we want to ensure the events are traveling the correct direction.
+
+
+### Every Path, Success And Failure
+
+We aim for 100% code coverage.  There's no real good reason for us to avoid it.  When there is a problem testing code that prevents us from achieving 100% coverage then we should correct the problem instead of skipping tests.
+
+Every true and every false condition should be tested.  Every promise should be resolved successfully and also should be rejected.  When considering options, make sure to handle the error scenarios as well as every type of success scenario.
+
+This may seem to contradict the "Test Behavior" section because we must be able to inspect the code to see every possible logical path that would be taken.  There is no conflict - each logical path results in outward behavior that the user might encounter.  Even if the code is later refactored to eliminate many logical paths, the tests should still pass and there would be no need to delete any of the tests that now follow a redundant path.
+
+
+Sample Test
+-----------
+
+This is a sample test for a directive in Angular.  Let's pretend our directive will simply set some property on an isolated scope to the value passed into the directive.  There are many things wrong with this test.
+
+    /*global angular, describe, expect, inject, it, module*/
+    describe('myCustomDirective', function () {
+        var $scope;
+
+        beforeEach(module('app'));
+        it('some property on scope', inject(function ($compile, $rootScope) {
```

Comment:
You're using single quotes here when the style guide for JavaScript says to use `double` quotes.